server/src/domain/domain.constant.ts

```diff
@@ -30,70 +30,73 @@ export function assertMachineLearningEnabled() {
+  '.pef': ['image/pef', 'image/x-pentax-pef'],
   '.png': 'image/png',
+  '.raf': ['image/raf', 'image/x-fuji-raf'],
+  '.raw': ['image/raw', 'image/x-panasonic-raw'],
+  '.rwl': ['image/rwl', 'image/x-leica-rwl'],
+  '.sr2': ['image/sr2', 'image/x-sony-sr2'],
+  '.srf': ['image/srf', 'image/x-sony-srf'],
+  '.srw': ['image/srw', 'image/x-samsung-srw'],
+  '.tiff': 'image/tiff',
   '.webp': 'image/webp',
+  '.x3f': ['image/x3f', 'image/x-sigma-x3f'],
 };
 
-const image: Record<string, string> = {
-  ...profile,
-  '.3fr': 'image/x-hasselblad-3fr',
-  '.ari': 'image/x-arriflex-ari',
-  '.arw': 'image/x-sony-arw',
-  '.cap': 'image/x-phaseone-cap',
-  '.cin': 'image/x-phantom-cin',
-  '.cr2': 'image/x-canon-cr2',
-  '.cr3': 'image/x-canon-cr3',
-  '.crw': 'image/x-canon-crw',
-  '.dcr': 'image/x-kodak-dcr',
-  '.erf': 'image/x-epson-erf',
-  '.fff': 'image/x-hasselblad-fff',
-  '.gif': 'image/gif',
-  '.iiq': 'image/x-phaseone-iiq',
-  '.k25': 'image/x-kodak-k25',
-  '.kdc': 'image/x-kodak-kdc',
-  '.mrw': 'image/x-minolta-mrw',
-  '.nef': 'image/x-nikon-nef',
-  '.orf': 'image/x-olympus-orf',
-  '.ori': 'image/x-olympus-ori',
-  '.pef': 'image/x-pentax-pef',
-  '.raf': 'image/x-fuji-raf',
-  '.raw': 'image/x-panasonic-raw',
-  '.rwl': 'image/x-leica-rwl',
-  '.sr2': 'image/x-sony-sr2',
-  '.srf': 'image/x-sony-srf',
-  '.srw': 'image/x-samsung-srw',
-  '.tiff': 'image/tiff',
-  '.x3f': 'image/x-sigma-x3f',
-};
+const profileExtensions = ['.avif', '.dng', '.heic', '.heif', '.jpeg', '.jpg', '.png', '.webp'];
+const profile: Record<string, string | string[]> = Object.fromEntries(
+  Object.entries(image).filter(([key]) => profileExtensions.includes(key)),
+);
 
-const video: Record<string, string> = {
+const video: Record<string, string | string[]> = {
   '.3gp': 'video/3gpp',
-  '.avi': 'video/x-msvideo',
+  '.avi': ['video/avi', 'video/msvideo', 'video/vnd.avi', 'video/x-msvideo'],
   '.flv': 'video/x-flv',
+  '.m2ts': 'video/mp2t',
   '.mkv': 'video/x-matroska',
   '.mov': 'video/quicktime',
-  '.mp2t': 'video/mp2t',
   '.mp4': 'video/mp4',
-  '.mpeg': 'video/mpeg',
+  '.mpg': 'video/mpeg',
+  '.mts': 'video/mp2t',
   '.webm': 'video/webm',
   '.wmv': 'video/x-ms-wmv',
 };
 
-const sidecar: Record<string, string> = {
-  '.xmp': 'application/xml',
+const sidecar: Record<string, string | string[]> = {
+  '.xmp': ['application/xml', 'text/xml'],
 };
 
-const isType = (filename: string, lookup: Record<string, string>) => !!lookup[extname(filename).toLowerCase()];
-const getType = (filename: string, lookup: Record<string, string>) => lookup[extname(filename).toLowerCase()];
+const isType = (filename: string, r: Record<string, string | string[]>) => extname(filename).toLowerCase() in r;
+
 const lookup = (filename: string) =>
-  getType(filename, { ...image, ...video, ...sidecar }) || 'application/octet-stream';
+  [{ ...image, ...video, ...sidecar }[extname(filename).toLowerCase()]].flat().at(0) ?? 'application/octet-stream';
```

Comment:
This looks pretty ugly to me. Can we instead change the objects above to Record<string, string[]> and move the array flattening stuff?

Reply:
Can you also change `??` back to `||`. I prefer and use `||` over `??` when I don't consider `false` or `''` as valid values.

Reply:
> Can you also change `??` back to `||`. I prefer and use `||` over `??` when I don't consider `false` or `''` as valid values.

Given the value can never be false or '', I think the intent behind `??` is clearer.

Reply:
> This looks pretty ugly to me. Can we instead change the objects above to Record<string, string[]> and remove the array flattening stuff?

Yeah I suppose we can. I just thought it might be nice for flexibility. Though uniformity is probably better.